### ISSUE TITLE
Regression model overfitting

### ISSUE THREAD

**Issue by Agnes-L** [guaguabujianle/MGraphDTA#6]
Thanks for sharing codes of MGraphDTA. I have successfully trained the models. However, when I ran the codes on my own data, I found that the overfitting problem was a bit severe in regression models. Could you please advise on which hyperparameters are more effective in suppressing model overfitting?

**Comment by guaguabujianle** [comment]
There are two possible solutions to solve the problem. First, you can increase dropout rate in nn.Dropout(). The current dropout rate is 0.1. May be you can try 0.2 or 0.5. Second, you can set weight_decay to a non-zero value (e.g., 1e-4 or 1e-5) in optim.Adam such as optim.Adam(model.parameters(), lr=5e-4, weight_decay=1e-4). Let me know if you have any further questions.

**Comment by Agnes-L** [comment]
Thanks for your kind help. I will train on my own data as soon as possible and discuss with you again if I have questions.

**Comment by Agnes-L** [comment]
Hello, I have run the code with the adjusted hyperparameters. However, it seems that the overfitting problem still exists, e.g. the difference in loss between the training and test sets is 0.4. I think it may also be related to my own data. I later used your split 'warm' and random sampling respectively. I found that the warm.kfold worked better but didn't look up the exact meaning in the text, so please would like to ask what the exact division of split='warm' is? Thanks for your kind help.

**Comment by guaguabujianle** [comment]
"warm" means warm-start split, which is similar to random split. In contrast, cold-start drug means that the datasets are split based  on drug, which means that drugs in the validation/test set are not available in the training set. Therefore, it's reasonable that the models show poor performance in the cold-start scenario.
